### PR TITLE
[ledger-api] - Rename max deduplication time to max deduplication duration [KVL-1311]

### LINKED ISSUE #12126: rename max_deduplication_time
The `max_deduplication_time` field name in the `LedgerConfiguration` is misleading. With the new command deduplication semantics, this is denotes the supported length of the deduplication period. For Daml 2.0, we want to rename this (without going through explicit deprecation) so that we avoid the confusion.

Suggested new name: `deduplication_period_length`.

### PR DESCRIPTION
**Pull request by nicu-da**
changelog_begin
ledger-api - rename ledger configuration max_deduplication_time to max_deduplication_duration
changelog_end
closes #12126 
- for consistency in the API

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala [hunk 1]
```diff
@@ -79,7 +79,7 @@ final class CommandsValidator(
         )
       deduplicationPeriod <- validateDeduplicationPeriod(
         commands.deduplicationPeriod,
-        maxDeduplicationTime,
+          maxDeduplicationDuration,
```
Comment (hubert-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala:82]: format?

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitAndWaitRequestValidator.scala [hunk 2]
```diff
@@ -31,7 +31,7 @@ class SubmitAndWaitRequestValidator(
         commands,
         currentLedgerTime,
         currentUtcTime,
-        maxDeduplicationTime,
+          maxDeduplicationDuration,
```
Comment (hubert-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitAndWaitRequestValidator.scala:34]: as above

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitRequestValidator.scala [hunk 3]
```diff
@@ -30,7 +30,7 @@ class SubmitRequestValidator(
         commands,
         currentLedgerTime,
         currentUtcTime,
-        maxDeduplicationTime,
+          maxDeduplicationDuration,
```
Comment (hubert-da) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitRequestValidator.scala:33]: as above

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitAndWaitRequestValidator.scala (hunk 2 above)
Comment (SamirTalwar-DA) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitAndWaitRequestValidator.scala:34]: Weird indentation.

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitRequestValidator.scala (hunk 3 above)
Comment (SamirTalwar-DA) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/SubmitRequestValidator.scala:33]: Here too.

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndexSpec.scala [hunk 4]
```diff
@@ -73,17 +73,17 @@ final class LedgerConfigurationSubscriptionFromIndexSpec
         offset("000a") -> Configuration(
           generation = 3,
           timeModel = LedgerTimeModel.reasonableDefault.copy(maxSkew = Duration.ofMinutes(1)),
-          maxDeduplicationTime = Duration.ofDays(1),
+            maxDeduplicationDuration = Duration.ofDays(1),
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndexSpec.scala:76]: And in this file.

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala (hunk 1 above)
Comment (SamirTalwar-DA) [inline ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala:82]: Oh, also here.

**Comment by SamirTalwar-DA** [comment]
_Note:_ this is a breaking change that we have agreed upon, as part of Daml 2.0.

**Review by SamirTalwar-DA — APPROVED** [review]

File: ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/apiserver/services/admin/ApiConfigManagementServiceSpec.scala [hunk 5]
```diff
@@ -108,7 +108,7 @@ class ApiConfigManagementServiceSpec
       val initialConfiguration = Configuration(
         generation = initialGeneration,
         timeModel = initialTimeModel,
-        maxDeduplicationTime = maximumDeduplicationTime,
+        maxDeduplicationDuration = maximumDeduplicationTime,
```
Comment (hubert-da) [inline ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/apiserver/services/admin/ApiConfigManagementServiceSpec.scala:111]: What about the right-hand side? Maybe it's also not worth naming the arguments.

File: ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/apiserver/services/admin/ApiConfigManagementServiceSpec.scala [hunk 6]
```diff
@@ -119,7 +119,7 @@ class ApiConfigManagementServiceSpec
       val expectedConfiguration = Configuration(
         generation = expectedGeneration,
         timeModel = expectedTimeModel,
-        maxDeduplicationTime = maximumDeduplicationTime,
+        maxDeduplicationDuration = maximumDeduplicationTime,
```
Comment (hubert-da) [inline ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/apiserver/services/admin/ApiConfigManagementServiceSpec.scala:122]: as above

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndexSpec.scala (hunk 4 above)
Comment (hubert-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/LedgerConfigurationSubscriptionFromIndexSpec.scala:76]: format here and below

File: ledger/participant-state/kvutils/tools/integrity-check/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/RawPreExecutingCommitStrategySupportSpec.scala [hunk 7]
```diff
@@ -94,7 +94,7 @@ class RawPreExecutingCommitStrategySupportSpec
         configuration = Configuration(
           generation = 1,
           timeModel = LedgerTimeModel.reasonableDefault,
-          maxDeduplicationTime = Duration.ofMinutes(1),
+            maxDeduplicationDuration = Duration.ofMinutes(1),
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/integrity-check/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/RawPreExecutingCommitStrategySupportSpec.scala:97]: format

File: ledger/participant-state/kvutils/tools/integrity-check/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/RawPreExecutingCommitStrategySupportSpec.scala [hunk 8]
```diff
@@ -106,7 +106,7 @@ class RawPreExecutingCommitStrategySupportSpec
         configuration = Configuration(
           generation = 2,
           timeModel = LedgerTimeModel.reasonableDefault,
-          maxDeduplicationTime = Duration.ofMinutes(1),
+            maxDeduplicationDuration = Duration.ofMinutes(1),
```
Comment (hubert-da) [inline ledger/participant-state/kvutils/tools/integrity-check/src/test/scala/ledger/participant/state/kvutils/tools/integritycheck/RawPreExecutingCommitStrategySupportSpec.scala:109]: format

File: ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeReadService.scala [hunk 9]
```diff
@@ -38,7 +38,7 @@ class BridgeReadService(
         config = Configuration(
           generation = 1L,
           timeModel = LedgerTimeModel.reasonableDefault,
-          maxDeduplicationTime = maximumDeduplicationDuration,
+            maxDeduplicationDuration = maximumDeduplicationDuration,
```
Comment (hubert-da) [inline ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/BridgeReadService.scala:41]: format

**Review by hubert-da — APPROVED** [review]
Can't tell if the changes are complete but, apart from formatting changes that you will have to do anyway, LGTM!

File: ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala (hunk 1 above)
Comment (nicu-da) [reply ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/CommandsValidator.scala:82]: hmmm, weird indeed, I run the format script right before pushing

File: docs/source/app-dev/command-deduplication.rst [hunk 10]
```diff
@@ -82,7 +82,7 @@ Independently of how the outcome is communicated, command deduplication generate
   The fields ``longest_duration`` or ``earliest_offset`` in the metadata specify the longest duration or earliest offset that is currently supported on the Ledger API endpoint.
   At least one of the two fields is present.
 
-  Neither deduplication durations up to the :ref:`maximum deduplication time <com.daml.ledger.api.v1.LedgerConfiguration.max_deduplication_time>` nor deduplication offsets published within that time SHOULD result in this error.
+  Neither deduplication durations up to the :ref:`maximum deduplication time <com.daml.ledger.api.v1.LedgerConfiguration.max_deduplication_duration>` nor deduplication offsets published within that time SHOULD result in this error.
```
Comment (andreaslochbihler-da) [inline docs/source/app-dev/command-deduplication.rst:85]: If we rename the field, we should also rename the references to it.
```suggestion
  Neither deduplication durations up to the :ref:`maximum deduplication duration <com.daml.ledger.api.v1.LedgerConfiguration.max_deduplication_duration>` nor deduplication offsets published within that time SHOULD result in this error.
```

**Pull request merged**